docs/syntax-jsx-support.md

```diff
@@ -0,0 +1,338 @@
+
+Mitigations:
+
+- Restrict JSX recognition to positions where a `simpleexp` is parsed.
+- Keep JSX syntax small initially (no raw text children, fragments, or spreads).
+- Continue gating behind a flag to allow collecting real-world compatibility data.
+
+### Editor integration & tooling considerations
+
+Adding JSX affects:
+
+- Tokenization/highlighting for `.luax`
+- Parsing for autocomplete, formatter/pretty-printer, and incremental parse
+- Error recovery expectations (JSX introduces new “paired delimiter” structures)
+
+Mitigations:
+
+- Keep JSX AST lowering to existing nodes so downstream tools operate on familiar AST shapes.
+- Ensure error recovery always makes progress to avoid hangs in partial code states.
+- Treat `.luax` as a distinct language mode/extension for editors.
+
+### Security / sandboxing considerations
+
+The syntax itself is not inherently unsafe, but it makes calling a factory function extremely easy. Sandboxing concerns are the same as for ordinary function calls:
+
+- The host controls which globals exist (including `jsx`).
+- Environments can restrict or replace `jsx` to safe implementations.
+
+### Performance considerations
+
+- Parsing adds new branches and lookahead, but is expected to be negligible compared to overall parsing cost.
+- Desugaring into existing AST nodes avoids adding new runtime node kinds and keeps later stages (typechecking, codegen) unchanged.
+
+### Status
+
+**Implemented (flagged / experimental)** in this repository as a spike:
+
+- `FFlag::LuauJsx` gating
+- `.luax` extension used to enable `ParseOptions::allowJsx`
+- Parser desugars JSX to `jsx(tag, props, ...children)`
+- Parser tests cover collision edge cases and error recovery
+
+This RFC is still required before any user-facing rollout.
+
+## Drawbacks
+
+- Adds a significant new surface area to the language syntax, which increases learning and maintenance cost.
+- JSX has a history of subtle grammar ambiguities; even with careful gating, future syntax additions may conflict.
+- Editor/tooling support must be updated (highlighting, formatting, autocomplete), and mismatches can degrade UX.
+- The “factory function” convention (`jsx`) is another community contract to standardize (or allow configuring).
+
+## Alternatives
+
+1. **Do nothing**: keep using `createElement(...)` calls and helper builders.
+2. **Library-level DSL**: implement an embedded DSL using Lua syntax only (functions/tables/metatables). This avoids parser changes but is typically more verbose and harder to read.
+3. **Transpile step**: introduce a `.luax -> .luau` transformer. This conflicts with the “interpreted/no build step” constraint and complicates tooling.
+4. **New AST node kinds**: keep JSX nodes in the AST and handle them in later passes. This can improve tooling fidelity but increases implementation complexity across compiler/typechecker/codegen and makes sandboxing harder to reason about.
```

Comment:
The luau parser supports parsing not just the AST but also the CST. Not having dedicated AST nodes for JSX would likely be problematic as we would lose necessary syntactic information required to for a clean round-trip